GVFS/GVFS.Common/Tracing/JsonTracer.cs

```diff
@@ -70,21 +71,28 @@ public bool HasLogFileEventListener
         public void AddEventListener(EventListener listener)
         {
             this.listeners.Add(listener);
+
+            // Tell the new listener about others who have previously failed
+            foreach (var kvp in this.failedListeners)

```

Comment:
We have not allowed `var` in the source code.  There are a couple other places where you are using it.